### DISCUSSION TITLE
Breaking change "make the eval consistent with function definition"

### DISCUSSION THREAD

**Discussion by forgedhallpass** [go-rod/rod#606] [General]
In the `formatToJSFunc()` function, the `detectJSFunction` condition branch was deleted. This assumes that all the JS code passed in is wrapped in a function. We are using `rod` for headless template execution, and I wonder if this change was intentional, because it breaks some of our templates.

e.g. https://github.com/projectdiscovery/nuclei-templates/blob/f88b297d1db9e75e6f88bfb8cd12e81bc96d3638/headless/extract-urls.yaml

Related commit: https://github.com/go-rod/rod/commit/edc1d6d7417e9e36eb124cc1f6bf3cc24f9187df
Current code: https://github.com/go-rod/rod/blame/0bb4269931964dbe05906aadf685d01cd5fcd979/page_eval.go#L113

**Comment by ysmood** [comment]
https://github.com/go-rod/rod/releases/tag/v0.104.0

**Reply by forgedhallpass** [reply]
Seems to be a strange decision, but alright. Thanks for the confirmation.

**Reply by ysmood** [reply]
Because too many people will miss it, and it has a lot of edge-case bugs.
